unstructured_ingest/v2/processes/connectors/notion/connector.py

```diff
@@ -0,0 +1,320 @@
+from dataclasses import dataclass
+from time import time
+from typing import Any, Generator, List, Optional, Set, Tuple
+
+from pydantic import Field, SecretStr
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+from unstructured_ingest.v2.processes.connectors.notion.client import Client as NotionClient
+from unstructured_ingest.v2.processes.connectors.notion.helpers import (
+    extract_database_html,
+    extract_page_html,
+    get_recursive_content_from_database,
+    get_recursive_content_from_page,
+)
+
+NOTION_API_VERSION = "2022-06-28"
+CONNECTOR_TYPE = "notion"
+
+
+class NotionConnectionConfig(AccessConfig):
+    notion_api_key: SecretStr = Field(description="Notion API key")
```

Comment:
The connection config should have an access config field on it. This doesn't match that pattern and will break the process. Take a look at any of the other v2 connectors, they all share the same pattern when it comes to the config interfaces.

Reply:
@rbiseck3 Do you have an example? All the connectors I could found on v2 were using the same approach

Reply:
For example: [airtable.py](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/airtable.py#L46-L55)
They all follow the pattern of:
```python
from unstructured_ingest.v2.interfaces import AccessConfig, ConnectionConfig
from pydantic import Secret

class MyAccessConfig(AccessConfig):
    sensitive_value: str

class MyConnectionConfig(ConnectionConfig):
    access_config: Secret[MyAccessConfig]
```

Reply:
@rbiseck3 On it!

Reply:
@rbiseck3 Addressed